integration/test/api_test.go

```diff
@@ -0,0 +1,242 @@
+			expectStatus: 200,
+		},
+		{
+			name:         "GET /v0/blob/{id} for unknown ID is 404",
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000",
+			expectStatus: 404,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000",
+			expectStatus: 500,
+			skip:         true,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000",
+			expectStatus: 503,
+			skip:         true,
+		},
+		{
+			name:         "GET /v0/blob/{id}/status is 200",
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/" + testBlobResp.ID + "/status",
+			expectStatus: 200,
+		},
+		{
+			name:         "GET /v0/blob/{id}/status for unknown ID is 404",
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000/status",
+			expectStatus: 404,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000/status",
+			expectStatus: 500,
+			skip:         true,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000/status",
+			expectStatus: 503,
+			skip:         true,
+		},
+	}
+
+	// Read and parse openapi.yaml for ensuring all paths, methods, and status
+	// codes are covered
+
+	schemaString, err := os.ReadFile("../../openapi.yaml")
+	require.NoError(t, err, "could not find openapi.yaml")
+
+	schemaMap := make(map[string]interface{})
+	err = yaml.Unmarshal(schemaString, schemaMap)
+	require.NoError(t, err)
+
+	var schemaCases []schemaCase
+
+	type kvmap = map[string]interface{}
+	for pathName, path := range schemaMap["paths"].(kvmap) {
+		for methodName, method := range path.(kvmap) {
+			for statusCode := range method.(kvmap)["responses"].(kvmap) {
+				schemaCases = append(schemaCases, schemaCase{
+					method:  methodName,
+					path:    pathName,
+					status:  statusCode,
+					covered: false,
+				})
+			}
+		}
+	}
+
+	// Run all tests
+	for _, test := range tests {
+		if !test.skip {
+			req, err := http.NewRequest(
+				test.onMethod,
+				requireJoinUrlPath(t, env.MotionAPIEndpoint, test.onPath),
+				bytes.NewReader([]byte(test.expectBody)),
+			)
+			req.Header.Set("Content-Type", test.onContentType)
+			require.NoError(t, err)
+
+			resp, err := http.DefaultClient.Do(req)
+			require.NoError(t, err)
+
+			// Body must be as expected
+			var body string
+			if test.expectBody != "" {
+				bodyBytes, err := io.ReadAll(resp.Body)
+				require.NoError(t, err)
+
+				body = string(bodyBytes)
+				require.Regexp(t, test.expectBody, string(body))
+			}
+
+			resp.Body.Close()
+
+			// Status code must be as expected
+			require.Equal(t, test.expectStatus, resp.StatusCode, "Incorrect status code for test %#v (resp body: %v)", test, body)
```

Comment:
I recommend using
https://github.com/parnurzeal/gorequest
to simplify above code

Reply:
I would avoid extra dependencies if we can help it.

Reply:
avoiding extra dependencies was my thinking while writing this

Reply:
I think such dependency would greatly reduce the LOC for testing, plus, it won't be built into the binary.